src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -733,12 +733,26 @@ public void EmitCFICodes(int offset)
             List<byte[]> cfis;
             if (_offsetToCfis.TryGetValue(offset, out cfis))
             {
-                foreach (byte[] cfi in cfis)
+                if (forArm)
                 {
-                    if (forArm)
+                    // Unwind insts are generated in the object file in the reversed order on arm,
+                    // so we should reverse the cfi list
+                    List<byte[]> reversed = new List<byte[]>();
+                    foreach (byte[] cfi in cfis)
+                    {
+                        reversed.Insert(0, cfi);
+                    }
+                    foreach (byte[] cfi in reversed)
```

Comment:
Maybe we can remove the reversing by iterating from back to front? `for (int index = cfis.Length - 1; index >= 0; index--) EmitARMExIdxCode(offset, cfis[index]);`

Reply:
@MichalStrehovsky Thanks, I'll try it

Reply:
@MichalStrehovsky I got the message
```
Compiler/DependencyAnalysis/ObjectWriter.cs(738,43): error CS1061: 'List<byte[]>' does not contain a definition for 'Length' and no extension method 'Length' accepting a first argument of type 'List<byte[]>' could be found (are you missing a using directive or an assembly reference?) [/share/CORERT/corert/src/ILCompiler.Compiler/src/ILCompiler.Compiler.csproj]
```

Reply:
Try `Count` instead of `Length`. GitHub UI doesn't have IntelliSense and I rely on it too much...